pandas/core/indexes/datetimelike.py

```diff
@@ -602,6 +604,17 @@ def delete(self, loc):
         arr = type(self._data)._simple_new(new_i8s, dtype=self.dtype, freq=freq)
         return type(self)._simple_new(arr, name=self.name)
 
+    @doc(Index._convert_arr_indexer)
+    def _convert_arr_indexer(self, keyarr):
+        if lib.infer_dtype(keyarr) == "string":
+            # Weak reasoning that indexer is a list of strings
+            # representing datetime or timedelta or period
+            extension_arr = pd_array(keyarr, self.dtype)
```

Comment:
if we have strings that cant be parsed to self.dtype this raises DateParseError (at least in the Period[D] case).  should we catch and re-raise as KeyError?

Reply:
@jbrockmendel 
Okay, It makes sense. I add exception handling and test code